EDDiscovery/UserControls/UserControlCompass.cs

```diff
@@ -0,0 +1,168 @@
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ * 
+ * EDDiscovery is not affiliated with Frontier Developments plc.
+ */
+using EliteDangerousCore;
+using EliteDangerousCore.JournalEvents;
+using System;
+using System.Drawing;
+using static EliteDangerousCore.DB.SQLiteDBClass;
+using static System.Math;
+
+namespace EDDiscovery.UserControls
+{
+    public partial class UserControlCompass : UserControlCommonBase
+    {
+
+        private HistoryEntry last_he;
+        private string DbLatSave { get { return "CompassLatTarget" + ((displaynumber > 0) ? displaynumber.ToString() : ""); } }
+        private string DbLongSave { get { return "CompassLongTarget" + ((displaynumber > 0) ? displaynumber.ToString() : ""); } }
+
+        #region Init
+
+        public UserControlCompass()
+        {
+            InitializeComponent();
+        }
+
+        public override void Init()
+        {
+            discoveryform.OnNewEntry += Display;
+            textBoxTargetLatitude.Text = GetSettingString(DbLatSave, "");
+            textBoxTargetLongitude.Text = GetSettingString(DbLongSave, "");
+        }
+
+        public override void Closing()
+        {
+            PutSettingString(DbLatSave, textBoxTargetLatitude.Text);
+            PutSettingString(DbLongSave, textBoxTargetLongitude.Text);
+            discoveryform.OnNewEntry -= Display;
+        }
+
+        #endregion
+
+        #region Display
+
+        public override Color ColorTransparency { get { return Color.Green; } }
+        public override void SetTransparency(bool on, Color curcol)
+        {
+            labelExt1.BackColor = labelExtBearing.BackColor = labelExtCurrLat.BackColor = labelExtTargetLong.BackColor = labelTargetLat.BackColor = curcol;
+            textBoxCurrentLatitude.BackColor = textBoxCurrentLongitude.BackColor = textBoxTargetLatitude.BackColor = textBoxTargetLongitude.BackColor = curcol;
+            BackColor = curcol;
+            Display();
+        }
+
+        public override void InitialDisplay()       // on start up, this will have an empty history
+        {
+            
+            Display(discoveryform.history.GetLast, discoveryform.history);
+        }
+        
+        private void Display(HistoryEntry he, HistoryList hl)
+        {
+            last_he = he;
+            Display();
+        }
+        
+        private void Display()
+        {
+            if (last_he != null)
+            {
+                double? latitude = null;
+                double? longitude = null;
+                switch (last_he.journalEntry.EventTypeID)
+                {
+                    case JournalTypeEnum.Screenshot:
+                        JournalScreenshot js = last_he.journalEntry as JournalScreenshot;
```

Comment:
What about adding a keys sequences to take "blind screenshots"? I mean, take a screenshot just for the purpose of get some coordinates, than purge it exactly after that... it could bypass the screenshot managing, to avoid mistakenly renaming, processing and movement of the same.

Reply:
I did think about an 'auto-delete screenshots taken while I'm open' option and might add one later.

The coordinates come from the journal so it needs to be a screen shot taken in the game which we can't drive from EDD - we could have a control option to send the screenshot key press to the game and auto-delete the next screen shot but that feels like overkill to me.

Reply:
Well, indeed it's not a trivial task. Anyway, I'm very confident about that panel.

Reply:
@iainross, maybe it could be awesome to have both proposed solution available to the user, which can select between on of them. The second one seems overkill but it could be the preferred approach for some user: if you are driving and find some interesting place to take a screenshot, and do not remember to close the compass, you'll find that you'll lose that magic moment... ,)